app/src/pages/inside/logsPage/modals/attachments/attachmentImageModal.jsx

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright 2019 EPAM Systems
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Component } from 'react';
+import PropTypes from 'prop-types';
+import Parser from 'html-react-parser';
+import track from 'react-tracking';
+import { injectIntl } from 'react-intl';
+import classNames from 'classnames/bind';
+import RotateImage from 'common/img/rotate-inline.svg';
+import { COMMON_LOCALE_KEYS } from 'common/constants/localization';
+import { ModalLayout, withModal } from 'components/main/modal';
+import { ATTACHMENT_IMAGE_MODAL_ID } from 'controllers/log/attachments';
+import { LOG_PAGE_EVENTS } from 'components/main/analytics/events';
+import { messages } from './messages';
+import styles from './attachmentImageModal.scss';
+
+const cx = classNames.bind(styles);
+
+@withModal(ATTACHMENT_IMAGE_MODAL_ID)
+@track()
+@injectIntl
+export class AttachmentImageModal extends Component {
+  static propTypes = {
+    data: PropTypes.shape({
+      image: PropTypes.string,
+    }).isRequired,
+    intl: PropTypes.object.isRequired,
+    tracking: PropTypes.shape({
+      trackEvent: PropTypes.func,
+      getTrackingData: PropTypes.func,
+    }).isRequired,
+  };
+
+  state = {
+    rotationAmount: 0,
+  };
+
+  getImageSize = () =>
+    this.imageRef && this.imageRef.current
+      ? {
+          height: this.imageRef.current.clientHeight,
+          width: this.imageRef.current.clientWidth,
+        }
+      : null;
+
+  getRotatedImageStyle = () => {
+    const { rotationAmount } = this.state;
+    this.initNativeImageSize();
+    const imageStyle = {
+      transform: this.generateRotationCommand(rotationAmount),
+    };
+
+    if (this.nativeImageSize) {
+      imageStyle.height =
+        (rotationAmount / 90) % 2 === 1
+          ? `${this.nativeImageSize.width}px`
+          : `${this.nativeImageSize.height}px`;
+      if (this.nativeImageSize.width < this.nativeImageSize.height) {
+        imageStyle.width =
+          (rotationAmount / 90) % 2 === 1
+            ? `${this.nativeImageSize.height}px`
+            : `${this.nativeImageSize.width}px`;
+      }
+    }
+    return imageStyle;
+  };
+
+  initNativeImageSize = () => {
```

Comment:
It looks like you need memoization here. This allows you to avoid creating new redundant class property `nativeImageSize` and also won't trigger an unnecessary reflow process. And then you can exclude calculation from the component class.

Reply:
Added to Tech Debt as a low priority fix.